src/Compression/CompressionCodecFSST.cpp

```diff
@@ -0,0 +1,145 @@
+#pragma GCC diagnostic ignored "-Wcast-align"
+#pragma GCC diagnostic ignored "-Wcast-qual"
+#pragma GCC diagnostic ignored "-Wold-style-cast"
+#pragma GCC diagnostic ignored "-Wimplicit-fallthrough"
+
+#include "config.h"
+
+#if USE_FSST
+
+#include <Compression/CompressionFactory.h>
+#include <Compression/CompressionInfo.h>
+#include <Compression/ICompressionCodec.h>
+#include <IO/VarInt.h>
+#include "base/types.h"
+
+#include <cstddef>
+#include <cstdint>
+#include <cstring>
+#include <stdexcept>
+#include <fsst.h>
+
+namespace DB
+{
+
```

Comment:
For my understanding: Why are we using the original FSST version and not the FSST12 variant?